core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -200,7 +210,101 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
   }
 
   @Override
-  public void close() throws IOException {
+  public void createNamespace(Namespace namespace, ImmutableMap<String, String> meta) {
+    Preconditions.checkArgument(
+        !namespace.isEmpty(),
+        "Cannot create namespace with invalid name: %s", namespace);
+    if (!meta.isEmpty()) {
+      throw new UnsupportedOperationException("Cannot create namespace " + namespace + " : metadata is not supported");
+    }
+
+    Path nsPath = new Path(warehouseLocation, SLASH.join(namespace.levels()));
+    FileSystem fs = Util.getFs(nsPath, conf);
```

Comment:
This file system should always be the same one. Can you add a final instance field for it using the warehouse path instead of calling `getFs` in each method?

Reply:
OK